src/field/pokemon.ts

```diff
@@ -4285,6 +4271,40 @@ export abstract class Pokemon extends Phaser.GameObjects.Container {
       return false;
     }
   }
+
+  /**
+   * Calculates the STAB multiplier for a move hitting this {@linkcode Pokemon}
+   * @param source the attacking {@linkcode Pokemon}
+   * @param move the {@linkcode Move} used in the attack
+   * @param abilityApplyMode the {@linkcode AbilityApplyMode} determining how abilities are applied.
+   * @param simulated If `true`, suppresses changes to game state during the calculation.
+   * @returns A {@linkcode NumberHolder} containing the STAB multiplier as value
+   */
+  public calcStabMultiplierForTakingDamage(
+    source: Pokemon,
+    move: Move,
+    abilityApplyMode: AbilityApplyMode,
+    simulated: boolean,
+  ): number {
```

Comment:
```suggestion
  ): number {
    if (move.hasAttr(TypelessAttr)) {
      return 1;
    }

```

A band-aid fix for #792. (`TypelessAttr` doesn't actually set a move's type to `ElementalType.UNKNOWN`, so every part of the codebase treats Struggle as a Normal-type move, which is its own can of worms.)